lightkurve/targetpixelfile.py

```diff
@@ -190,6 +196,122 @@ def from_archive(target, cadence='long', quarter=None, month=None,
+            Right ascension [degrees]
+        DEJ2000: astropy.table.column (KIC & EPIC)
+            Declination [deg]
+        pmRA: astropy.table.column (KIC & EPIC)
+            Proper motion for right ascension [mas/year]
+        e_pmRA: astropy.table.column (KIC & EPIC)
+            Error proper motion for Right ascension [mas/year]
+        pmDEC: astropy.table.column (KIC & EPIC)
+            Proper motion for declination [mas/year]
+        e_pmDEC: astropy.table.column (EPIC)
+            Error proper motion for declination [mas/year]
+        Kpmag: astropy.table.column (KIC & EPIC)
+            Magnitude in Kepler band [mag]
+        """
+
+        if catalog is None:
+            if self.mission == 'Kepler':
+                catalog = 'KIC'
+            elif self.mission == 'K2':
+                catalog = 'EPIC'
+            else:
+                log.error('Please provide a catalog.')
+
+        if catalog is "Gaia":
+            log.warn('Gaia RAs and Decs are at EPOC 2015.5. These RA/Decs have not been corrected.')
+
+        # Vizier id's
+        ID = {"KIC":
+                    {'vizier':"V/133/kic",
+                     'parameters':["KIC", "RAJ2000", "DEJ2000", "pmRA", "pmDE", "kepmag"]},
+               "EPIC":
+               {'vizier':"IV/34/epic",
+                'parameters':["ID", "RAJ2000", "DEJ2000", "pmRA", "pmDEC", "e_pmRA", "e_pmDEC", "Kpmag"]},
+               "Gaia":
+               {'vizier':"I/345/gaia2",
+               'parameters':["DR2Name", "RA_ICRS", "DE_ICRS", "pmRA", "pmDE", "e_pmRA", "e_pmDE", "Gmag"]}}
+
+        # identifies catalog
+        viz_id = (ID[catalog])['vizier']
+        pars = (ID[catalog])['parameters']
+
+        #Skycoord the centre of target
+        cent = SkyCoord(ra=self.ra, dec=self.dec, frame='icrs', unit=(u.deg, u.deg))
+
+        # Choose columns from Vizier
+        v = Vizier(catalog=[viz_id], columns=pars)
+        # query around cent. with radius
+        result = v.query_region(cent, radius=radius*u.arcmin, catalog=viz_id)
+
+        # Rename names in KIC for consistency with EPIC
+        result[viz_id]['KIC'].name = 'ID'
+        result[viz_id]['pmDE'].name = 'pmDEC'
+        result[viz_id]['kepmag'].name = 'Kpmag'
+
+        # Queried stats
+        data = result[viz_id]
+
+        # Load ra & dec of all tpf pixels
+        gc_ra, gc_dec = self.get_coordinates(cadence='all')
+
+        # Reduce calculation for astroy seperation
+        c1 = np.asarray([gc_ra.ravel(), gc_dec.ravel()])
+        co = np.round(c1, decimals=5)
+
+        # Return unique pairs
+        pairs_gc = (np.unique(co, axis=1))
+
+        # Make pairs_gc into SkyCoord
+        sky_pairs_gc = SkyCoord(ra=pairs_gc[0]*u.deg, dec=pairs_gc[1]*u.deg, frame='icrs', unit=(u.deg, u.deg))
+        # Make pairs in sky_data
+        sky_data = SkyCoord(ra=data['RAJ2000'], dec=data['DEJ2000'], frame='icrs', unit=(u.deg, u.deg))
+
+        new_ra, new_dec = [], []
```

Comment:
All of this can be saved by doing a boolean array and then masking the `data` table.